system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -3485,7 +3488,7 @@ class CLdapClient : implements ILdapClient, public CInterface
             TIMEVAL timeOut = {m_ldapconfig->getLdapTimeout(),0};
 
             Owned<ILdapConnection> lconn = m_connections->getConnection();
-            LDAP* ld = lconn.get()->getLd();
+            ld = lconn.get()->getLd();
```

Comment:
Should this code also use ld if it is passed in?

Reply:
It could and I suppose would be consistent. My main concern was as little impact as possible. However, I will make the change to use the passed in connection if present.

Reply:
After some further testing, I remember why this path was not changed. When creating a user for a non Active Directory type server, the password is set when the user is created. Therefore, the method is not called during user creation. It is only called when the password is updated, in which case the connection will not be valid. So, I left it the way it was. Sorry for the confusion with the previous comment. 

Please merge as is.

Reply:
Ideally there would be a comment in the code explaining this - so that the same issue isn't raised later.  I will merge as-is though.